.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,24 @@
+### Description
+
+_Please explain the changes you made here._
+_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
+_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
+
+----
+
+### Checklist (never delete this)
+
+Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below. 
+
+#### Content
+- [ ] Added or modified tests to cover changes and any new functionality
+- [ ] All tests passing on Travis 
+
+#### Form
+- [ ] Rebased, squashed and reworded to produce a single commit (exemptions may be made -- sparingly) that is _uncluttered by excess lines_ (lines that just say "addressed review", "fixed" etc left over from squashing)
```

Comment:
(I would argue it's not rocket science, but clearly reality proves me wrong on a daily basis)

The internet abounds with lists of what's important. I would boil it down to:

````
Write a short, descriptive and capitalized title

Start body after a blank line (critical; tools like rebase can get confused 
if you run the two together). Body contains more detailed explanatory 
text, if necessary. Further paragraphs come after blank lines. Wrap to
~72 characters if possible. 

- Bullet points are okay, too  
- Typically a hyphen or asterisk is used for the bullet, followed by a 
  single space, with blank lines in between, but conventions vary here  
- Use a hanging indent
````

Any additional pet peeves/requirements